packages/arb-token-bridge-ui/src/components/common/Dialog.tsx

```diff
@@ -6,9 +6,11 @@ import { twMerge } from 'tailwind-merge'
 import { Button, ButtonProps } from './Button'
 import { getTransitionProps } from './Transition'
 /**
- * Returns a promise which resolves to a boolean value, `false` if the action was canceled and `true` if it was confirmed.
+ * Returns a promise which resolves to an array [boolean, unknown] value,
+ * `false` if the action was canceled and `true` if it was confirmed.
+ * Second index contain any additional information
  */
-type WaitForInputFunction = () => Promise<boolean>
+type WaitForInputFunction = () => Promise<[boolean, unknown]>
```

Comment:
It was already updated everywhere. But if we prefer to have `boolean | [boolean, unknown]` as a return type, then we need to check for `Array.isArray(result)` everywhere we call `waitForInput`. Also, if someone forget to check for it, then if the result is `[false, data]`, it would be considered true